src/xdp/trn_transit_xdp.c

```diff
@@ -454,6 +457,23 @@ static __inline int trn_process_inner_ip(struct transit_packet *pkt)
 
 	__be64 tunnel_id = trn_vni_to_tunnel_id(pkt->geneve->vni);
 
+	/*
+	* If target ip is towards a virtual subnet, send it up to user space
+	*/
+	struct network_key_t nkey;
+	struct network_t *net;
+	nkey.prefixlen = 96;
+
+	__builtin_memcpy(&nkey.nip[0], &tunnel_id, sizeof(tunnel_id));
+	nkey.nip[2] = pkt->inner_ip->daddr;
+	net = bpf_map_lookup_elem(&networks_map, &nkey);
+	// send pkt to user space if this is the edge gateway and dest inner ip belongs to a virtual subnet
+	if ( net && net->virtual && pkt->ip->daddr == net->cluster_gateway) {
```

Comment:
please remove whitespace between '(' 'net' for consistent coding style. prefer explicit net !=0 &&net->virtual != 0) && ... as these are not bool.

Reply:
golang has lint (reformatting) and you can select to auto-format upon committing changes.

also net !=0 actually still can be confusing as to "why 0? what does 0 mean?". Probably add a #define for 0 with a meaningful name to make it clear what 0 means where.

Reply:
bpf_map_lookup_elem  returns void * which is NULL if not found. I added the check here.